kvbc/include/pruning_handler.hpp

```diff
@@ -189,4 +189,47 @@ class PruningHandler : public reconfiguration::IPruningHandler {
   mutable std::mutex pruning_status_lock_;
   mutable std::future<void> async_pruning_res_;
 };
+
+/*
+ * The read only pruning handler knows to answer only getLatestPruneableBlock.
+ * As it doesn't participant the consensus, the only thing we want to know is what is the latest reachable block.
+ */
+class ReadOnlyReplicaPruningHandler : public reconfiguration::IPruningHandler {
+ public:
+  ReadOnlyReplicaPruningHandler(IReader &ro_storage)
+      : ro_storage_{ro_storage},
+        signer_{bftEngine::ReplicaConfig::instance().replicaPrivateKey},
+        pruning_enabled_{bftEngine::ReplicaConfig::instance().pruningEnabled_},
+        replica_id_{bftEngine::ReplicaConfig::instance().replicaId} {}
+  bool handle(const concord::messages::LatestPrunableBlockRequest &,
+              concord::messages::LatestPrunableBlock &latest_prunable_block,
+              concord::messages::ReconfigurationErrorMsg &) override {
+    if (!pruning_enabled_) return true;
```

Comment:
I guess this might not be related to this PR, but I was wondering if we should return 0 as the last reachable block in case pruning is disabled. Does it make sense?

Reply:
Actually, now that I think of it, maybe a read-only replica doesn't even care if pruning is enabled. What do you think?

Reply:
We want not to touch anything in case pruning is disabled, because there might be other handlers that will be called (currently it is only concord's pruning handler, we want only one pruning handler (concord's) and for that we simply deactivate the internal command handler)

Reply:
I see, thanks. I was thinking that we might want to always return the last block for read-only replica, irrespective of `pruning_enabled_`.

Also, wanted to ask something else. What will the value be for `latest_prunable_block` if `pruning_enabled_ = false`?

Reply:
The point is that we may want to return something else from another pruning handler. So if the basic pruning (or any other pruning handler) is not enabled, we want not to touch the latest pruneable block as we may override a previous value of it. The value will be the one that set to the reference by either another pruning handler or by the dispatcher.